src/libsync/propagatedownload.cpp

```diff
@@ -176,7 +172,6 @@ void GETFileJob::slotMetaDataChanged()
         // Disable the buffer limit, as we don't limit the bandwidth for error messages.
         // (We are only going to do a readAll() at the end.)
         reply()->setReadBufferSize(0);
-        _device->close();
```

Comment:
Is that safe? 

As the comment say, we do not want to write to the file because we need to keep the data for error reporting.

Reply:
See commit message: I think it's safe because there's an extra `_saveBodyToFile` flag that's only set to true if `slotMetaDataChanged` doesn't detect any error.